anthony-green/scripts/articleView.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+let articleView = {};
+
+articleView.populateFilters = () => {
+  $('article').each(function() {
+    if (!$(this).hasClass('template')) {
+      let val = $(this).find('address a').text();
+      let optionTag = `<option value="${val}">${val}</option>`;
+
+      if ($(`#author-filter option[value="${val}"]`).length === 0) {
+        $('#author-filter').append(optionTag);
+      }
+
+      val = $(this).attr('data-category');
+      optionTag = `<option value="${val}">${val}</option>`;
+      if ($(`#category-filter option[value="${val}"]`).length === 0) {
+        $('#category-filter').append(optionTag);
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = () => {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-author="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = () => {
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      $(`article[data-category="${$(this).val()}"]`).fadeIn();
+    } else {
+      $('article').fadeIn();
+      $('article.template').hide();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = () => {
+  $('.main-nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+    $(`#${$(this).data('content')}`).fadeIn();
+  });
+
+  $('.main-nav .tab:first').click();
+};
+
+articleView.setTeasers = () => {
+  $('.article-body *:nth-of-type(n+2)').hide();
+  $('article').on('click', 'a.read-on', function(e) {
+    e.preventDefault();
+    if ($(this).text() === 'Read on →') {
+      $(this).parent().find('*').fadeIn();
+      $(this).html('Show Less &larr;');
+    } else {
+      $('body').animate({
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+// COMMENT: Where is this function called? Why?
+// This is called after the other function calls. 
+articleView.initNewArticlePage = () => {
+  // TODO: Ensure the main .tab-content area is revealed. We might add more tabs later or otherwise edit the tab navigation.
+  $('tab-content').show();
+
+  // TODO: The new articles we create will be copy/pasted into our source data file.
+  // Set up this "export" functionality. We can hide it for now, and show it once we have data to export.
+
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
```

Comment:
when something is changed in the field, you will need to call articleView.create()